### PR TITLE
added Table._filter_partitions

### PR DESCRIPTION
**Pull request by cseed**
Fixes https://github.com/hail-is/hail/issues/3619

FYI @konradjk

### CODE REVIEW

File: src/main/scala/is/hail/table/Table.scala [hunk 1]
```diff
@@ -901,4 +901,19 @@ class Table(val hc: HailContext, val tir: TableIR) {
 
     copy2(rvd = newRVD, signature = newRVType)
   }
+
+  def filterPartitions(parts: java.util.ArrayList[Int], keep: Boolean): MatrixTable =
```
Comment (jbloom22) [inline src/main/scala/is/hail/table/Table.scala]: should return Table

File: src/main/scala/is/hail/table/Table.scala [hunk 2]
```diff
@@ -901,4 +901,19 @@ class Table(val hc: HailContext, val tir: TableIR) {
...
+    filterPartitions(parts.asScala.toArray, keep)
+
+  def filterPartitions(parts: Array[Int], keep: Boolean = true): MatrixTable = {
```
Comment (jbloom22) [inline src/main/scala/is/hail/table/Table.scala]: same

File: src/main/scala/is/hail/rvd/UnpartitionedRVD.scala [hunk 3]
```diff
@@ -91,5 +91,13 @@ class UnpartitionedRVD(val rowType: TStruct, val crdd: ContextRDD[RVDContext, Re
     OrderedRVD.shuffle(ordType, newPartitioner, filtered)
   }
 
+  def subsetPartitions(keep: Array[Int]): OrderedRVD = {
+    require(keep.length <= crdd.partitions.length, "tried to subset to more partitions than exist")
+    require(keep.isIncreasing && (keep.isEmpty || (keep.head >= 0 && keep.last < crdd.partitions.length)),
```
Comment (jbloom22) [inline src/main/scala/is/hail/rvd/UnpartitionedRVD.scala]: I think it makes more sense to only have the second require here and where you copied this from, with the word "sorted" replaced by "increasing".

Comment (cseed) [reply src/main/scala/is/hail/rvd/UnpartitionedRVD.scala]: I left the first assert (cheap, good error message) but changed sorted to increasing.

**Pull request merged**